src/index.js

```diff
@@ -12,13 +13,41 @@ class CancelRun extends Error {
   }
 }
 
+async function runTestsSerial(tests, watcher, onStart, onResult, onFailure) {
+  const runner = async test => {
+    return onStart(test).then(() => {
+      return new Promise((resolve, reject) => {
+        lint(
+          { testPath: test.path, config: test.context.config },
+          (err, result) => {
+            if (err) {
+              reject(err);
+            } else {
+              resolve(result);
+            }
+          },
+        );
+      });
+    });
+  };
+  return Promise.all(
+    tests.map(test =>
```

Comment:
Do you want these to be serial, or single-threaded?

This approach is single-threaded, but is also concurrent. If you wanted serial execution, you'd need to reduce over `tests` instead of using Promise.all + map.

The function name should match the implementation :-)